proposals/testing/NNNN-targeted-interoperability-swift-testing-and-xctest.md

```diff
@@ -0,0 +1,311 @@
+
+```swift
+func assertUnique(_ elements: [Int]) {
+  XCTAssertEqual(Set(elements).count, elements.count, "\(elements) has non unique elements")
+}
+
+// XCTest
+
+class FooTests: XCTestCase {
+    func testDups() {
+        assertUnique([1, 2, 1]) // Fails as expected
+    }
+}
+
+// Swift Testing
+
+@Test func `Duplicate elements`() {
+  assertUnique([1, 2, 1]) // Passes? Oh no!
+}
+```
+
+Generally, you encounter the above limitation with testing APIs when _all_ the
+following conditions are met:
+
+- You call XCTest API in a Swift Testing test, or call Swift Testing API in a
+  XCTest test,
+- The API doesn't function as expected in some or all cases, and
+- You get no notice at build time or runtime about the malfunction
+
+For the remainder of this proposal, we’ll describe test APIs which exhibit this
+limitation as **lossy without interop**.
+
+You could regress test coverage if you migrate to Swift Testing without
+replacing usage of lossy without interop test APIs. Furthermore, you may want to
+ensure you don't inadvertently introduce new XCTest API after completing your
+Swift Testing migration.
+
+## Proposed solution
+
+- **XCTest APIs which are lossy without interop will work as expected when
+  called in Swift Testing.** In addition, runtime diagnostics will be included
+  to highlight opportunities to adopt newer Swift Testing equivalent APIs.
+
+- Conversely, **Swift Testing API called in XCTest will work as expected if
+  XCTest already provides similar functionality.** You should always feel
+  empowered to choose Swift Testing when writing new tests or test helpers, as
+  it will work properly in both types of tests.
+
+We don't propose supporting interoperability for APIs which are not lossy
+without interop, because they naturally have high visibility. For example, using
+`throw XCTSkip` in a Swift Testing test results in a test failure rather than a
+test skip, providing a clear indication that migration is needed.
+
+## Detailed design
+
+> [!NOTE]
+> This proposal refers to XCTest in the abstract. There are two different
+> implementations of XCTest: the open source [Corelibs XCTest][] and a
+> [proprietary XCTest][Xcode XCTest] that is shipped as part of Xcode. The Swift
+> evolution process governs changes to the former only. Therefore,
+> this proposal is targeted for Corelibs XCTest.
+
+[Corelibs XCTest]: https://github.com/swiftlang/swift-corelibs-xctest
+[Xcode XCTest]: https://developer.apple.com/documentation/xctest
+
+### Highlight and support XCTest APIs which are lossy without interop
+
+We propose supporting the following XCTest APIs in Swift Testing:
+
+- [Assertions][XCTest Assertions]: `XCTAssert*` and [unconditional failure][]
+  `XCTFail`
+- [Expected failures][], such as `XCTExpectFailure`: marking a Swift Testing
+  issue in this way will generate a runtime warning issue.
+- [`XCTAttachment`][XCTest attachments]
+- [Issue handling traits][]: we will make our best effort to translate issues
```

Comment:
Is this point about _traits_ or about issue handling in general? (Since we're not translating traits.)